codec/codec-fury/src/main/java/com/alipay/sofa/rpc/codec/fury/FurySerializer.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.codec.fury;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.alipay.sofa.rpc.codec.AbstractSerializer;
+import com.alipay.sofa.rpc.common.RemotingConstants;
+import com.alipay.sofa.rpc.common.struct.ConcurrentHashSet;
+import com.alipay.sofa.rpc.common.utils.CodecUtils;
+import com.alipay.sofa.rpc.common.utils.StringUtils;
+import com.alipay.sofa.rpc.config.ConfigUniqueNameGenerator;
+import com.alipay.sofa.rpc.context.RpcInvokeContext;
+import com.alipay.sofa.rpc.core.exception.SofaRpcException;
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.ext.Extension;
+import com.alipay.sofa.rpc.transport.AbstractByteBuf;
+import com.alipay.sofa.rpc.transport.ByteArrayWrapperByteBuf;
+
+import io.fury.Fury;
+import io.fury.Language;
+
+/**
+ * @author lipan
+ */
+@Extension(value = "fury", code = 15)
+public class FurySerializer extends AbstractSerializer {
+
+    private final FuryHelper    furyHelper  = new FuryHelper();
+    private final Set<Class<?>> registerSet = new ConcurrentHashSet<>();
+    Fury                        fury;
+
+    public FurySerializer() {
+        fury = Fury.builder()
```

Comment:
There can refer to this configuration here.
```
Fury f = Fury.builder()
              .withLanguage(Language.JAVA)
              .withReferenceTracking(true)
              .withCodegen(true)
              .withNumberCompressed(true)
              .withCompatibleMode(CompatibleMode.COMPATIBLE)
              .withClassRegistrationRequired(false)
              .withClassLoader(classLoader)
              .withAsyncCompilationEnabled(true)
              .build();
```

Reply:
thanks, i will refer

Reply:
but,have some version, it not some method,like _withReferenceTracking_ and _withClassRegistrationRequired_
i think, this _withReferenceTracking_ has changed his name to _withRefTracking_, and then ,about _withClassRegistrationRequired_ is no such method anymore in the new version.

and then,whether there is a version requirement.

Reply:
There may be differences between versions. I contact Fury support to confirm these parameters.

Reply:
thanks,I looked at the code.
In the new version.
_withReferenceTracking_ rename to _withRefTracking_,
_withClassRegistrationRequired_ rename to _requireClassRegistration_

Reply:
are there any other code needs to be modified?

Reply:
OK